crates/red_knot_python_semantic/src/types.rs

```diff
@@ -3838,8 +3816,10 @@ pub(crate) mod tests {
     }
 
     #[test]
-    fn typing_vs_typeshed_no_default() {
-        let db = setup_db();
+    fn typing_vs_typeshed_no_default() -> anyhow::Result<()> {
+        let db = TestDbBuilder::new()
+            .with_python_version(PythonVersion::PY313)
+            .build()?;
```

Comment:
This is the only functional change, without any consequences. But it feels okay, since `typing.NoDefault` is only available since 3.13). I will need this — and similar things — soon in the statically-know branch, so I'm putting it into place already.

Reply:
I think once we have the statically-known branch, we will probably want to add more tests around some of these "exists in `typing` in recent versions and in `typing_extensions` as fallback" known symbols, in terms of how they behave on older and newer Pythons. But this change seems fine for this PR and this particular test.